http-server/src/main/java/io/airlift/http/server/HttpServer.java

```diff
@@ -139,7 +144,19 @@ public HttpServer(
             log.info("Virtual threads support is enabled");
             threadPool.setVirtualThreadsExecutor(executor);
         }
-        server = new Server(threadPool);
+
+        int maxBufferSize = toIntExact(max(
```

Comment:
Maybe ensure that minimum is `DEFAULT_MAX_CAPACITY` so you do not shrink it too much.